modules/oe_translation_poetry/src/Form/AddLanguagesRequestForm.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_translation_poetry\Form;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Url;
+
+/**
+ * Form for requesting new translations.
+ */
+class AddLanguagesRequestForm extends PoetryCheckoutFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'oe_translation_poetry_add_languages_request';
+  }
+
+  /**
+   * Returns the title of the form page.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $node
+   *   The node entity to use when generating the title.
+   *
+   * @return \Drupal\Core\StringTranslation\TranslatableMarkup
+   *   The title for the form page.
+   */
+  public function getPageTitle(ContentEntityInterface $node = NULL): TranslatableMarkup {
+    $queue = $this->queueFactory->get($node);
+    $entity = $queue->getEntity();
```

Comment:
We already have the node entity in the method parameters so there is no need to get it again.

Reply:
Not exactly. The queue provides us with the exact revision of the entity. And its title is the one we want to use.